src/_macosx.m

```diff
@@ -664,6 +659,7 @@ static CGFloat _get_device_scale(CGContextRef cr)
 FigureManager_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
     Window* window = [Window alloc];
+    ++FigureWindowCount;
```

Comment:
Yep...moved it to later so it won't increment if there was an error and the window did not actually get created.